src/modules/Instabug.ts

```diff
@@ -99,29 +101,24 @@ function _logFlags() {
  * @param config SDK configurations. See {@link InstabugConfig} for more info.
  */
 export const init = async (config: InstabugConfig) => {
-  // Initialize necessary variables
-  isNativeInterceptionFeatureEnabled = await NativeNetworkLogger.isNativeInterceptionEnabled();
   if (Platform.OS === 'android') {
-    hasAPMNetworkPlugin = await NativeNetworkLogger.hasAPMNetworkPlugin();
-    shouldEnableNativeInterception =
-      config.networkInterceptionMode === NetworkInterceptionMode.native;
-  }
-
-  // Add app state listener to handle background/foreground transitions
-  addAppStateListener(async (nextAppState) => handleAppStateChange(nextAppState, config));
+    // Add android feature flags listener for android
+    addOnFeatureUpdatedListener(config);
+  } else {
+    isNativeInterceptionFeatureEnabled = await NativeNetworkLogger.isNativeInterceptionEnabled();
 
-  // Perform platform-specific checks and update interception mode
-  handleNetworkInterceptionMode(config);
+    // Add app state listener to handle background/foreground transitions
+    addAppStateListener(async (nextAppState) => handleAppStateChange(nextAppState, config));
```

Comment:
TBH I don't think listening manually on app state change (background/foreground) is a good idea for making sure that the feature flags are in sync for the same reason as why the approach didn't work on Android. I think it'd be better to follow the same approach with a listener like on Android to avoid the feature flags getting out of sync between native and JS.

Reply:
@a7medev, AWU, however this implementation is done due to and iOS limitation.